apps/arweave/src/ar_verify_chunks.erl

```diff
@@ -299,6 +321,59 @@ report_progress(State) ->
 	ar_verify_chunks_reporter:update(StoreID, Report2),
 	State#state{ verify_report = Report2 }.
 
+%% Generate a list of Count random offsets in the range (Start, End]
+%% (i.e. offsets strictly greater than Start and less than or equal to End)
+%% such that if an offset is sampled, no other offsets are selected from the
+%% open interval (Offset - ?DATA_CHUNK_SIZE, Offset + ?DATA_CHUNK_SIZE).
+generate_sample_offsets(Start, End, Count) when is_integer(Start), is_integer(End) ->
+	Candidates = lists:seq(Start + 1, End, ?DATA_CHUNK_SIZE),
```

Comment:
For a full 3.6TB partition I think Candidates will be a list of length 14.4 million entries or have I misread? 

Could that become a memory issue (e.g. if running verify on multiple storage modules concurrently)?